### PR TITLE
CI config to notify internal CI.

### PR DESCRIPTION
**Pull request by yunikkk**
Same as Android's https://github.com/mapbox/mapbox-maps-android/pull/1621. 
Will send notifications to the internal CI when main or release branches are updated

## Pull request checklist:
 - [ ] Describe the changes in this PR, especially public API changes.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.
 - [ ] Add any new public, top-level symbols to the Jazzy config's `custom_categories` (scripts/doc-generation/.jazzy.yaml)
 - [ ] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] Update the guides (internal access only), README.md, and DEVELOPING.md if their contents are impacted by these changes.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` first and then port to `v10.[version]` release branch.

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -190,6 +190,13 @@ workflows:
             branches:
               only: main
 
+      # Notify internal CI about main and release branch updates
```
Comment (yunikkk) [inline .circleci/config.yml:193]: @evil159 @OdNairy would like your input whether `steve` is right workflow to run this `trigger` job

Comment (OdNairy) [reply .circleci/config.yml:193]: Yep, that right.
`Steve` workflow is something we run on each PR/merge.

File: .circleci/config.yml [hunk 2]
```diff
@@ -916,6 +923,24 @@ jobs:
...
+      - run:
+          name: Notify maps internal CI about $CIRCLE_BRANCH branch update
+          command: |
+            curl -X POST "https://circleci.com/api/v2/project/github/mapbox/mapbox-maps-ios-private/pipeline" \
+                -H 'Content-Type: application/json' \
+                -H "Circle-Token: $CIRCLECI_API_TOKEN_MAPBOX_MAPS_IOS_INTERNAL" \
+                -d $'{
+                        "branch": "main",
+                        "parameters": {
+                          "mapbox_sync_branch": "$CIRCLE_BRANCH"
```
Comment (OdNairy) [inline .circleci/config.yml:940]: I have some light concerns about $CIRCLE_BRANCH. Most of the time we would trigger build system with the `CIRCLE_BRANCH=main`. It means that potentially, there are might be parallels builds in the `ios-private` repo. 
Could it be an issue?

**Review by OdNairy — APPROVED** [review]
LGTM

**Pull request merged**